### ISSUE TITLE
Override release timeout per-request

### ISSUE THREAD

**Issue by proegssilb** [sLoPPydrive/PGPool#4]
RocketMap probably wants to lease for a couple hours, while PGNumbra probably only wants accounts for 15 minutes. Certainly, while I've been working with PGNumbra, I've wanted to be able to set the release timeout to 5 minutes.

This should be as simple as adding an optional parameter when requesting accounts (and an extra DB column to explicitly keep track of request expiration). Default value is whatever is specified in the config, or the application can specify said parameter and request accounts for a different duration.

This feature can permit a Denial of Service attack, but that can be mitigated with an authenticating proxy, or by using the default value as an "upper bound" on the permitted release timeout. Or by adding an extra config setting for controlling default/max release timeout independently.

**Comment by sLoPPydrive** [comment]
Usually you don't need to care about the lease timeout. PGNumbra requests all `--pgpool_num_accounts` accounts at once from the pool where they get the system ID `pgnumbra_<pid>`. When PGNumbra is done with an account it releases it actively to the pool so that it becomes available again immediately. You don't need to wait for it to time out.

You may ask what happens if the release timeout is shorter than the PGNumbra total runtime. Well, that's currently a (minor) issue indeed because accounts might get released by the timeout and used again by clients under certain circumstances before PGNumbra gets to them to test them. I should add batch processing so that not all accounts get requested at once.

**Comment by TheQL** [comment]
While batch processing generally sounds like a good idea most accounts that are able to be assigned to workers should not be targeted for checking by PGNumbra in the first place.

**Comment by sLoPPydrive** [comment]
This is already the case as PGNumbra only picks bad accounts or accounts in unknown state. ;)

**Comment by TheQL** [comment]
That's what I was trying to say, thus the theoretically existing race condition with release timeout and worker assigning overlapping should not happen. PGNumbra picks different accounts than workers would.

**Comment by proegssilb** [comment]
I've had a number of cases where PGNumbra crashes, and fails to release the accounts it claimed. Because I'm using PGNumbra on an isolated cloud machine, this means I wind up tearing down the instance, doing something else for two hours, and coming back to it later, only to repeat the same song and dance. The person actually running the PGPool instance is unresponsive.

Frankly, I've been dealing with other stuff, or I'd look into writing a PR for this myself.

**Comment by TheQL** [comment]
So your problem is actually that you have PGNumbra's system ID assigned to accounts and that won't go away and also you don't have direct access to the DB to just manually set it to null again?

**Comment by proegssilb** [comment]
Yes, but that wouldn't be a problem if the release timeout was controllable per-claim.
